test/testcafe/framework/page-objects/components/CalloutObject.js

```diff
@@ -13,22 +15,24 @@ export default class CalloutObject {
   }
 
   isWarning() {
+    if (userVariables.gen3) {
+      return this.el.hasClass('MuiAlert-calloutWarning');
+    }
     return this.el.hasClass('infobox-warning') &&
       this.el.child('[data-se="icon"]').hasClass('warning-16');
   }
 
   isError() {
-    const hasInfoBoxErrorClass = this.el.hasClass('infobox-error');
     if (userVariables.gen3) {
-      return hasInfoBoxErrorClass;
+      return this.el.hasClass('MuiAlert-calloutError');
     }
-    return hasInfoBoxErrorClass &&
+    return this.el.hasClass('infobox-error') &&
       this.el.find('[data-se="icon"]').hasClass('error-16');
```

Comment:
This is sort of random but I'm a bit skeptical of `error-16` and `warning-16`. Are these `emotion` classes? I guess this has been working for gen2 tests but it seems like the presence of `'infobox-error` should be enough to tell us that the callout is the error variant

Reply:
These are not emotion classes. They are used in Gen2. `16` means width and height of icon. For Gen3 we should use `infobox-*` data-se attribute

Reply:
Ah okay I see, thanks for the clarification